test/new-e2e/tests/windows/base_agent_installer_suite.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package windows contains the code to run the e2e tests on Windows
+package windows
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	platformCommon "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/windows/common"
+	windowsAgent "github.com/DataDog/datadog-agent/test/new-e2e/tests/windows/common/agent"
+	infraCommon "github.com/DataDog/test-infra-definitions/common"
+	"path/filepath"
+	"strings"
+)
+
+// BaseAgentInstallerSuite is a base class for the Windows Agent installer suites
+type BaseAgentInstallerSuite[Env any] struct {
+	e2e.BaseSuite[Env]
+
+	AgentPackage *windowsAgent.Package
+	OutputDir    string
+}
+
+// InstallAgent installs the Agent on a given Windows host. It will pass all the parameters to the MSI installer.
+func (b *BaseAgentInstallerSuite[Env]) InstallAgent(host *components.RemoteHost, options ...windowsAgent.InstallAgentOption) (string, error) {
+	b.T().Helper()
+	p, err := infraCommon.ApplyOption(&windowsAgent.InstallAgentParams{
+		InstallLogFile: "install.log",
+	}, options)
+	if err != nil {
+		return "", err
+	}
+
+	if p.Package == nil {
+		return "", fmt.Errorf("missing agent package to install")
+	}
+
+	var args []string
+
+	if p.AgentUser != "" {
+		args = append(args, fmt.Sprintf("DDAGENTUSER_NAME=%s", p.AgentUser))
+	}
+	if p.AgentUserPassword != "" {
+		args = append(args, fmt.Sprintf("DDAGENTUSER_PASSWORD=%s", p.AgentUserPassword))
+	}
+	if p.Site != "" {
+		args = append(args, fmt.Sprintf("SITE=%s", p.Site))
+	}
+	if p.APIKey != "" {
+		args = append(args, fmt.Sprintf("APIKEY=%s", p.APIKey))
+	}
+	if p.DdURL != "" {
+		args = append(args, fmt.Sprintf("DD_URL=%s", p.DdURL))
+	}
```

Comment:
Does it make sense to add a function to `InstallAgentParams` to generate the arg string?